src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionFixAllResolveHandler.cs

```diff
@@ -0,0 +1,84 @@
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Newtonsoft.Json.Linq;
+using Roslyn.Utilities;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using LSP = Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler.CodeActions
+{
+    [ExportCSharpVisualBasicStatelessLspService(typeof(CodeActionFixAllResolveHandler)), Shared]
+    [Method("codeAction/resolveFixAll")]
+    internal class CodeActionFixAllResolveHandler : ILspServiceDocumentRequestHandler<RoslynFixAllCodeAction, RoslynFixAllCodeAction>
+    {
+        private readonly ICodeFixService _codeFixService;
+        private readonly ICodeRefactoringService _codeRefactoringService;
+        private readonly IGlobalOptionService _globalOptions;
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public CodeActionFixAllResolveHandler(
+            ICodeFixService codeFixService,
+            ICodeRefactoringService codeRefactoringService,
+            IGlobalOptionService globalOptions)
+        {
+            _codeFixService = codeFixService;
+            _codeRefactoringService = codeRefactoringService;
+            _globalOptions = globalOptions;
+        }
+
+        public bool MutatesSolutionState => false;
+
+        public bool RequiresLSPSolution => true;
+
+        public TextDocumentIdentifier GetTextDocumentIdentifier(RoslynFixAllCodeAction request)
+            => ((JToken)request.Data!).ToObject<CodeActionResolveData>()!.TextDocument;
+
+        public async Task<RoslynFixAllCodeAction> HandleRequestAsync(RoslynFixAllCodeAction request, RequestContext context, CancellationToken cancellationToken)
+        {
+            var document = context.GetRequiredDocument();
+            var data = ((JToken)request.Data!).ToObject<CodeActionResolveData>();
+            Assumes.Present(data);
+
+            var options = _globalOptions.GetCodeActionOptionsProvider();
+
+            var codeActions = await CodeActionHelpers.GetCodeActionsAsync(
+                document,
+                data.Range,
+                options,
+                _codeFixService,
+                _codeRefactoringService,
+                cancellationToken,
+                request.Scope).ConfigureAwait(false);
+
+            var codeActionToResolve = CodeActionHelpers.GetCodeActionToResolve(data.UniqueIdentifier, codeActions);
+            Contract.ThrowIfNull(codeActionToResolve);
+
+            var fixAllCodeAction = (FixAllCodeAction)codeActionToResolve;
```

Comment:
how do you know this cast is safe?

Reply:
Given that this point in the code would only be reached if the user selects a Fix All action from the list.